## Import Data

In [1]:
import pandas as pd

In [2]:
google_sheet_id = '10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw'
url = 'https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
monthly_expenses = pd.read_excel(url, sheet_name ='Sheet1' )

In [3]:
google_sheet_id = '1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4'
url = 'https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
payroll = pd.read_excel(url, sheet_name ='Sheet1' )

In [4]:
google_sheet_id = '1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c'
url = 'https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
daily_marketing_spendings = pd.read_excel(url, sheet_name ='Sheet1' )

In [5]:
google_sheet_id = '1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE'
url = 'https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
payments = pd.read_excel(url, sheet_name ='Sheet1' )

In [6]:
google_sheet_id = '1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI'
url = 'https://docs.google.com/spreadsheets/d/' + google_sheet_id + '/export?format=xlsx'
lifespan = pd.read_excel(url, sheet_name ='Sheet1' )

##1. CAC: Customer Acquisition Cost


##1.1. Software Marketing

In [7]:
monthly_expenses.sample(5)


#      month           category                  item  amount
17  18 2023-02-01              Other       Travel Expenses    3100
22  23 2023-03-01  Software Licenses            Salesforce    1700
10  11 2023-02-01       Server Costs  Google Cloud Storage    4200
3    4 2023-01-01  Software Licenses                 Slack     800
15  16 2023-02-01      Office Rental           Office Rent   10000

In [8]:
expenses_202303= monthly_expenses[monthly_expenses['month'] == '2023-03-01']

In [9]:
crm_expenses = expenses_202303[expenses_202303['item'] == 'Salesforce']['amount'].values[0]
crm_expenses

np.int64(1700)

##1.2. Salaries

In [10]:
payroll.sample(5)

month department employee_name            position  paid
34 2023-03-01      Sales      John Doe       Sales Manager  1500
49 2023-03-01  Analytics  Sarah Wilson        Data Analyst  1200
35 2023-03-01      Sales    Jane Smith     Sales Associate   600
13 2023-01-01    Support   Emily Clark  Support Specialist   450
32 2023-02-01  Analytics  Sarah Wilson        Data Analyst  1200

In [11]:
payroll_202303= payroll[payroll['month'] == '2023-03-01']

In [12]:
marketing_payroll= payroll_202303[(payroll_202303['department'] == 'Marketing') | (payroll_202303['department']== 'Sales')]['paid'].sum()
marketing_payroll

np.int64(5950)

##1.3. Daily Marketing Expenses

In [13]:
daily_marketing_spendings.sample(5)

date       channel  spending
44  2023-01-12    Google Ads       523
0   2023-01-01    Google Ads       784
190 2023-02-17  LinkedIn Ads       423
336 2023-03-26    Google Ads       440
329 2023-03-24  Facebook Ads       898

In [14]:
daily_marketing_spendings_202303 = daily_marketing_spendings[(daily_marketing_spendings['date'] >= '2023-03-01') & (daily_marketing_spendings['date'] <= '2023-03-31')]

In [15]:
daily_marketing_spendings = daily_marketing_spendings_202303['spending'].sum()
daily_marketing_spendings

np.int64(68830)

In [16]:
total_Acquisition_Cost = crm_expenses + marketing_payroll + daily_marketing_spendings
total_Acquisition_Cost

np.int64(76480)

##1.4. Number of custumers

In [17]:
payments.sample(5)

date  customer_id  receipt_amount  new_customer
224 2023-01-22         2457             366             0
902 2023-03-28         1370             140             0
352 2023-02-06         2731             242             0
19  2023-01-04         2345             277             0
773 2023-03-17         2528             164             1

In [18]:
payments_202303 = payments[(payments['date'] >= '2023-03-01') & (payments['date'] <= '2023-03-31')]

In [19]:
numbers_of_customer =payments_202303[payments_202303['new_customer'] == 1]['customer_id'].nunique()
numbers_of_customer

63

##1.5. CAC

In [20]:
cac = total_Acquisition_Cost / numbers_of_customer
cac

np.float64(1213.968253968254)

Kết luận và khuyến nghị:

Đơn vị chi phí thu hút khách hàng (CAC) của công ty hiện tại là khoảng 1.214 USD cho mỗi khách hàng mới trong tháng 3 năm 2023.

Tối ưu chi phí marketing và CRM để giảm CAC.

Nâng cao trải nghiệm, giữ chân khách hàng để tăng hiệu quả đầu tư.

#2. ARPU

##2.1. Total Revenue

In [21]:
total_revenue = payments_202303['receipt_amount'].sum()
total_revenue

np.int64(83033)

##2.2. Number of customers

In [22]:
numbers_of_customer = payments_202303['customer_id'].nunique()
numbers_of_customer

292

##2.3. ARPU Caculation

In [23]:
arpu = total_revenue / numbers_of_customer
arpu

np.float64(284.3595890410959)

Kết luận và kiến nghị:

ARPU hiện là 284 USD, thấp hơn nhiều so với CAC 1.214 USD. Công ty cần tập trung tăng ARPU bằng cách nâng cao giá trị khách hàng và đồng thời giảm chi phí thu hút khách để cải thiện lợi nhuận và bền vững.

#3. COGS

##3.1. Software expenses



In [24]:
production_software = ['AWS Hosting', 'Google Cloud Storage', 'Atlassian Jira']
production_expense = expenses_202303[expenses_202303['item'].isin(production_software)]['amount'].sum()
production_expense

np.int64(14200)

In [25]:
shared_software = ['Slack', 'Zoom']
shared_expense = expenses_202303[expenses_202303['item'].isin(shared_software)]['amount'].sum()*0.6
shared_expense

np.float64(864.0)

In [26]:
software_expenses = production_expense + shared_expense
software_expenses

np.float64(15064.0)

##3.2. Production salary

In [27]:
production_salary = payroll_202303[payroll_202303['department'] == 'Engineering']['paid'].sum()
production_salary

np.int64(5200)

##3.3. COGS Calculation

In [28]:
cogs = software_expenses + production_salary
cogs

np.float64(20264.0)

Kết luận và kiến nghị:

Chi phí sản xuất (COGS) của công ty là 20.264 USD, bao gồm chi phí phần mềm và lương nhân viên sản xuất. Đây là khoản chi phí quan trọng ảnh hưởng trực tiếp đến lợi nhuận. Công ty nên xem xét tối ưu hóa các chi phí này để nâng cao hiệu quả kinh doanh và tăng biên lợi nhuận.










#4. Gross Margin


In [29]:
gross_margin = (total_revenue - cogs) / total_revenue * 100
gross_margin


np.float64(75.5952452639312)

Kết luận và kiến nghị:

Biên lợi nhuận gộp của công ty đạt 75.6%, cho thấy công ty giữ lại được phần lớn doanh thu sau khi trừ đi chi phí sản xuất trực tiếp. Đây là dấu hiệu tích cực cho thấy hiệu quả trong quản lý chi phí sản xuất. Công ty nên tiếp tục duy trì hoặc nâng cao biên lợi nhuận này bằng cách tối ưu chi phí và tăng giá trị sản phẩm.

#5. LTV - Life tine Value

In [30]:
lifespan['lifespan_days'] = (lifespan['churn_date'] - lifespan['start_date']).dt.days
lifespan['lifespan_months'] = lifespan['lifespan_days'] / 30
average_lifespan_months = lifespan['lifespan_months'].mean()

In [31]:
ltv = arpu * average_lifespan_months * gross_margin
ltv

np.float64(211551.59315068496)

Kết luận và kiến nghị:

Giá trị vòng đời khách hàng trung bình là 211.551, phản ánh tổng doanh thu kỳ vọng thu được từ một khách hàng trong suốt thời gian họ gắn bó với công ty. Đây là một chỉ số quan trọng để đo lường hiệu quả dài hạn của các chiến dịch tiếp thị và chăm sóc khách hàng. Công ty nên tập trung nâng cao LTV bằng cách cải thiện trải nghiệm khách hàng và tăng khả năng giữ chân khách hàng.


#6. LVT/CAC


In [32]:
ltv / cac

np.float64(174.26451841648995)

Kết luận và kiến nghị:

Tỷ lệ LTV trên CAC rất cao, đạt 174.26, cho thấy công ty tạo ra giá trị khách hàng vượt xa chi phí bỏ ra để thu hút họ. Đây là dấu hiệu cho thấy mô hình kinh doanh rất hiệu quả và bền vững. Công ty nên duy trì tỷ lệ này và cân nhắc đầu tư thêm vào các hoạt động marketing để mở rộng quy mô.